new-lamassu-admin/src/pages/Transactions/DetailsCard.jsx

```diff
@@ -135,14 +132,10 @@ const DetailsRow = ({ it: tx, timezone }) => {
     Number.parseFloat(tx.commissionPercentage, 2) * 100
   ).toFixed(2, 1) // ROUND_DOWN
   const fixedFee = Number.parseFloat(tx.fixedFee) || 0
-  const fiat = BigNumber(tx.fiat)
-    .minus(fixedFee)
-    .toFixed(2, 1) // ROUND_DOWN
+  const fiat = BigNumber(tx.fiat).minus(fixedFee).toFixed(2, 1) // ROUND_DOWN
   const crypto = getCryptoAmount(tx)
   const cryptoFee = tx.fee ? `${getCryptoFeeAmount(tx)} ${tx.fiatCode}` : 'N/A'
-  const exchangeRate = BigNumber(fiat)
-    .div(crypto)
-    .toFixed(2, 1) // ROUND_DOWN
+  const exchangeRate = BigNumber(fiat).div(crypto).toFixed(2, 1) // ROUND_DOWN
```

Comment:
I know this is just UI code and that it was already there, but maybe we should avoid mixing float and bignum arithmetic. That `fiat`, for example, is converted to string, never to be used as a string (it's converted to a bignum again right after)...